src/CommitmentsV2/SFA.DAS.CommitmentsV2/Extensions/QueryableApprenticeshipsExtensions.cs

```diff
@@ -181,5 +186,102 @@ private static IQueryable<Apprenticeship> WithAlertsEmployer(this IQueryable<App
             return identifier.EmployerAccountId.HasValue ? 
                 apprenticeships.Where(app => app.Cohort.EmployerAccountId == identifier.EmployerAccountId) : apprenticeships;
         }
+
+        public static IQueryable<Apprenticeship> FilterApprenticeshipByAlert( IQueryable<Apprenticeship> apprenticeships, Alerts alert, long? providerId)
+        {
+            switch (alert)
+            {
+                case Alerts.IlrDataMismatch:
+                    return apprenticeships.Where(a =>
+                        (
+                            a.DataLockStatus.Any(x =>
+                                (x.ErrorCode.HasFlag(DataLockErrorCode.Dlock03)
+                                 || x.ErrorCode.HasFlag(DataLockErrorCode.Dlock04)
+                                 || x.ErrorCode.HasFlag(DataLockErrorCode.Dlock05)
+                                 || x.ErrorCode.HasFlag(DataLockErrorCode.Dlock06)) &&
+                                x.TriageStatus == TriageStatus.Unknown &&
+                                !x.IsResolved)
+                        ) ||
+                        (
+                            a.DataLockStatus.Any(x =>
+                                ((int) x.ErrorCode == (int) DataLockErrorCode.Dlock07) &&
+                                x.TriageStatus == TriageStatus.Unknown &&
+                                !x.IsResolved)
+                        )
+                    );
+                  
+                case Alerts.ChangesPending:
+                    return apprenticeships.Where(a =>
+                        (
+                            a.DataLockStatus.Any(x =>
+                                (
+                                    x.ErrorCode.HasFlag(DataLockErrorCode.Dlock03)
```

Comment:
There are extension methods which check for some of these combinations of flags you have here - they should be used instead for consistency

Reply:
The extension methods don't work for Linq to SQL had a lot of issues with them. Already had a look with Corey

Reply:
Yeah the existing extensions methods aren't compatible with Linq to SQL. The alternative was to moved some of the logic into methods which return Expression<Func<Apprenticeship, bool>> but it was not straight forward and could be addressed as a later tech debt.

Reply:
I think this should definitely be raised as a tech debt as the same logic is now being repeated multiple times - lets include the tech debt ticket number here before we resolve this comment

Reply:
Issue number --> https://skillsfundingagency.atlassian.net/browse/CON-3631